src/server/qgsrequesthandler.h

```diff
@@ -96,33 +96,26 @@ class QgsRequestHandler
     //! @note not available in Python bindings
     virtual void setGetCoverageResponse( QByteArray* ba ) = 0;
 
-    virtual void setDefaultHeaders() {}
-
     //! Set an HTTP header
     virtual void setHeader( const QString &name, const QString &value ) = 0;
 
     //! Remove an HTTP header
-    virtual int removeHeader( const QString &name ) = 0;
-
-    //! Delete all HTTP headers
-    virtual void clearHeaders() = 0;
+    virtual void removeHeader( const QString &name ) = 0;
 
     //! Append the bytestream to response body
     virtual void appendBody( const QByteArray &body ) = 0;
 
-    //! Clears the response body
-    virtual void clearBody() = 0;
```

Comment:
"BTW you can always get the list of headers keys and clear them one by one from a plugin":
As a matter af fact no :  this possibility did not even existed before (see the QgsRequestHandler interface in master) so there is little advantage to enable it.

In fact yes, there is a drawback: you need to implement it ! 
We may be hitting a philosophical point here :-). My point of view is: keep code minimal, if you don't see the use for a feature, do not implement it - until it is required.

Note that from now, the response parameters are abstracted in QgsServerResponse objects (an abstract QgsRequestHandler is no more needed) and I really would like to keep the its interface minimal.